google-cloud-examples/src/main/java/com/google/cloud/examples/pubsub/snippets/PublisherClientSnippets.java

```diff
@@ -0,0 +1,210 @@
+      ListTopicsRequest listTopicsRequest =
+          ListTopicsRequest.newBuilder()
+              .setProjectWithProjectName(ProjectName.create(projectId))
+              .setPageSize(100)
+              .setPageToken(pageToken)
+              .build();
+      ListTopicsPagedResponse response =
+          publisherClient.listTopics(listTopicsRequest);
+      Iterable<Topic> topics = response.iterateAllElements();
+      for (Topic topic : topics) {
+        // do something with the topic
+      }
+      // [END listTopicsWithPageToken]
+      return response;
+    }
+  }
+
+  /** Example of listing topics for a subscription, specifying the page size. */
+  public ListTopicSubscriptionsPagedResponse listTopicSubscriptions(String topicName) throws Exception {
+    try (PublisherClient publisherClient = PublisherClient.create()) {
+      TopicName formattedTopicName = TopicName.create(projectId, topicName);
+      // [START listTopicSubscriptions]
+      ListTopicSubscriptionsRequest request =
+          ListTopicSubscriptionsRequest.newBuilder()
+              .setTopicWithTopicName(formattedTopicName)
+              .setPageSize(100)
+              .build();
+      ListTopicSubscriptionsPagedResponse response =
+          publisherClient.listTopicSubscriptions(request);
+      Iterable<String> subscriptions = response.iterateAllElements();
+      for (String subscription : subscriptions) {
+        // do something with the subscription name
+      }
+      // [END listTopicSubscriptions]
+      return response;
+    }
+  }
+
+  /** Example of listing topics for a subscription, specifying the page size and page token */
+  public ListTopicSubscriptionsPagedResponse listTopicSubscriptionsWithPageToken(
+      String topicName, String pageToken) throws Exception {
+    try (PublisherClient publisherClient = PublisherClient.create()) {
+      TopicName formattedTopicName = TopicName.create(projectId, topicName);
+      // [START listTopicSubscriptionsWithPageToken]
+      ListTopicSubscriptionsRequest request =
+          ListTopicSubscriptionsRequest.newBuilder()
+              .setTopicWithTopicName(formattedTopicName)
+              .setPageSize(100)
+              .setPageToken(pageToken)
+              .build();
+      ListTopicSubscriptionsPagedResponse response =
+          publisherClient.listTopicSubscriptions(request);
+      Iterable<String> subscriptions = response.iterateAllElements();
+      for (String subscription : subscriptions) {
+        // do something with the subscription name
+      }
+      // [END listTopicSubscriptionsWithPageToken]
+      return response;
+    }
+  }
+
+  /** Example of deleting a topic. */
+  public TopicName deleteTopic(String topicName) throws Exception {
+    try (PublisherClient publisherClient = PublisherClient.create()) {
+      TopicName formattedTopicName = TopicName.create(projectId, topicName);
+      // [START deleteTopic]
+      publisherClient.deleteTopic(formattedTopicName);
```

Comment:
`formattedTopicName` isn't the right name for a variable of the type `TopicName` (this seems to occur in many snippets above). Also, why isn't the line for creating the topic name object inside the snippet (also seems to occur in many of the snippets)?